### DISCUSSION TITLE
Webhook listener is crashing the app

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [Shopify/shopify-app-template-node#802] [Q&A]
Registered for the Webhook, added listener with a callback as well. On Product change getting below error. Could not find the root cause. How can we fix this? 

Failed to process webhook: Error: Could not validate request for topic products/update
┃ node:internal/errors:464
┃     ErrorCaptureStackTrace(err);
┃     ^
┃ 
┃ Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
┃     at new NodeError (node:internal/errors:371:5)
┃     at ServerResponse.setHeader (node:_http_outgoing:576:11)
┃     at ServerResponse.header (/Users/Workspace/rbridge/node_modules/express/lib/response.js:776:10)
┃     at ServerResponse.contentType (/Users/Workspace/rbridge/node_modules/express/lib/response.js:604:15)
┃     at ServerResponse.send (/Users/Workspace/rbridge/node_modules/express/lib/response.js:145:14)
┃     at file:///Users/Workspace/rbridge/server/index.js:65:23
┃     at processTicksAndRejections (node:internal/process/task_queues:96:5) {
┃   code: 'ERR_HTTP_HEADERS_SENT'

**Comment by JaKXz** [comment]
Hey @ramkishore-yerubandi thanks for posting -- we have a fix for this in progress: https://github.com/Shopify/shopify-app-node/pull/801 

You should be able to apply the fix locally with (assuming you're using git):

```
curl https://patch-diff.githubusercontent.com/raw/Shopify/shopify-app-node/pull/801.diff | git apply
git checkout -b fix-webhook-listener
```

**Reply by ramkishore-yerubandi** [reply]
I applied the fix. Though the app is not crashing anymore, and the webHookHandler is getting triggered...on console I am getting the below log:
 Webhook processed, returned status code 200
┃ Failed to process webhook: Error: Could not validate request for topic products/update
┃ Failed to process webhook: Error: Could not validate request for topic products/update

**Reply by JaKXz** [reply]
@nickwesselman did you encounter this in your exploration?

**Reply by nickwesselman** [reply]
@JaKXz This error would seem to indicate that the HMAC didn't validate, potentially an incorrect API Key in the app?

https://github.com/Shopify/shopify-node-api/blob/main/src/webhooks/registry.ts#L433
